File: server/pkg/api/game/game.go
package game

import (
	"scrabble/pkg/scrabble"
)

type Game struct {
	ID             string         `json:"id"`
	CreatorID      string         `json:"creatorId"`
	UserIDs        []string       `json:"userIds"`
	ObservateurIDs []string       `json:"observateurIds"`
	IsPrivateGame  bool           `json:"isPrivateGame"`
	IsProtected    bool           `json:"isProtected"`
	WinnerID       string         `json:"winnerId"`
	HashedPassword string         `json:"-"`
	ScrabbleGame   *scrabble.Game `json:"-"`
	StartTime      int64          `json:"startTime"`
	TournamentID   string         `json:"tournamentId"`
}

func (g *Game) IsTournamentGame() bool {
	return g.TournamentID != ""
}

func (g *Game) IsJoinable() bool {
	if g.ScrabbleGame != nil {
		return false
	}
	if g.TournamentID != "" {
		return false
	}
	return len(g.UserIDs) < 4
}

File: mobile/lib/models/game.dart
class Game {
  String id;
  String creatorId;
  List<String> userIds;
  List<String> observateurIds;
  bool isPrivateGame;
  bool isProtected;

  Game(
      {required this.id,
        required this.creatorId,
        required this.userIds,
        required this.observateurIds,
        required this.isPrivateGame,
        required this.isProtected
      });

  factory Game.fromJson(Map<String, dynamic> json) {
    int userIdsIndex = 0;
    int observateurIdsIndex = 0;
    return Game(
        id: json["id"],
        creatorId: json["creatorId"],
        userIds: List<String>.from((json["userIds"] as List).map(
                (userId) => json["userIds"][userIdsIndex++]
        )),
        observateurIds: List<String>.from((json["observateurIds"] != null
            ? json["observateurIds"] as List : []).map(
                (userId) => json["observateurIds"][observateurIdsIndex++]
        )),
        isPrivateGame: json["isPrivateGame"],
        isProtected: json["isProtected"]
    );
  }

  Map<String, dynamic> toJson() => {
    "id": id,
    "creatorId": creatorId,
    "userIds": userIds,
    "observateurIds": observateurIds,
    "isPrivateGame": isPrivateGame,
    "isProtected": isProtected
  };
}